opendbc/car/__init__.py

```diff
@@ -100,117 +92,6 @@ class Bus(StrEnum):
-
-  val = np.clip(val, min_lim, max_lim)
-
-  # slow rate if val increases in magnitude
-  if val_last > 0:
-    val = np.clip(val,
-               max(val_last - STEER_DELTA_DOWN, -STEER_DELTA_UP),
-               val_last + STEER_DELTA_UP)
-  else:
-    val = np.clip(val,
-               val_last - STEER_DELTA_UP,
-               min(val_last + STEER_DELTA_DOWN, STEER_DELTA_UP))
-
-  return float(val)
-
-
-def apply_meas_steer_torque_limits(apply_torque, apply_torque_last, motor_torque, LIMITS):
-  return int(round(apply_dist_to_meas_limits(apply_torque, apply_torque_last, motor_torque,
-                                             LIMITS.STEER_DELTA_UP, LIMITS.STEER_DELTA_DOWN,
-                                             LIMITS.STEER_ERROR_MAX, LIMITS.STEER_MAX)))
-
-
-def apply_std_steer_angle_limits(apply_angle: float, apply_angle_last: float, v_ego: float, steering_angle: float,
-                                 lat_active: bool, limits: AngleSteeringLimits) -> float:
-  # pick angle rate limits based on wind up/down
-  steer_up = apply_angle_last * apply_angle >= 0. and abs(apply_angle) > abs(apply_angle_last)
-  rate_limits = limits.ANGLE_RATE_LIMIT_UP if steer_up else limits.ANGLE_RATE_LIMIT_DOWN
-
-  angle_rate_lim = np.interp(v_ego, rate_limits[0], rate_limits[1])
-  new_apply_angle = np.clip(apply_angle, apply_angle_last - angle_rate_lim, apply_angle_last + angle_rate_lim)
-
-  # angle is current steering wheel angle when inactive on all angle cars
-  if not lat_active:
-    new_apply_angle = steering_angle
-
-  return float(np.clip(new_apply_angle, -limits.STEER_ANGLE_MAX, limits.STEER_ANGLE_MAX))
-
-
-def common_fault_avoidance(fault_condition: bool, request: bool, above_limit_frames: int,
-                           max_above_limit_frames: int, max_mismatching_frames: int = 1):
-  """
-  Several cars have the ability to work around their EPS limits by cutting the
-  request bit of their LKAS message after a certain number of frames above the limit.
-  """
-
-  # Count up to max_above_limit_frames, at which point we need to cut the request for above_limit_frames to avoid a fault
-  if request and fault_condition:
-    above_limit_frames += 1
-  else:
-    above_limit_frames = 0
-
-  # Once we cut the request bit, count additionally to max_mismatching_frames before setting the request bit high again.
-  # Some brands do not respect our workaround without multiple messages on the bus, for example
-  if above_limit_frames > max_above_limit_frames:
-    request = False
-
-  if above_limit_frames >= max_above_limit_frames + max_mismatching_frames:
-    above_limit_frames = 0
-
-  return above_limit_frames, request
-
-
-def apply_center_deadzone(error, deadzone):
-  if (error > - deadzone) and (error < deadzone):
-    error = 0.
-  return error
-
-
-def rate_limit(new_value, last_value, dw_step, up_step):
-  return float(np.clip(new_value, last_value + dw_step, last_value + up_step))
```

Comment:
this is not only used in lateral functions. maybe we also need a car/controls/__init__.py with common stuff

Reply:
I think the MVP is leaving it here and importing it on `lateral.py`, and a future improvement is having it in `car/controls/__init__.py`

Reply:
Im having second thoughts on this, I will make the `car/controls/` and move `lateral.py` there and make the `__init__.py` it kinda feels like the right thing to do, but let's see how it looks

Reply:
Alright I've done this, but I didn't really find what else we should put on `car/controls/__init__.py` that's not too disruptive. I found `car/common/pid.py` and I could put it on `car/controls/` but not sure if it's what we want? or putting the contents of `pid.py` inside the `__init__.py` but that would be like going backwards maybe?